content/en/docs/File Format/implementationstatus.md

```diff
@@ -0,0 +1,101 @@
+---
+title: "Implementation status"
+linkTitle: "Implementation status"
+weight: 8
+---
+### Physical types
+
+| Data type                                 | C++   | Java   | Go    | Rust  |
+| ----------------------------------------- | ----- | ------ | ----- | ----- |
+| BOOLEAN                                   |       |        |       |       |
+| INT32                                     |       |        |       |       |
+| INT64                                     |       |        |       |       |
+| INT96                                     |       |        |       |       |
+| FLOAT                                     |       |        |       |       |
+| DOUBLE                                    |       |        |       |       |
+| BYTE_ARRAY                                |       |        |       |       |
+| FIXED_LEN_BYTE_ARRAY                      |       |        |       |       |
+
+### Logical types
+
+| Data type                                 | C++   | Java   | Go    | Rust  |
+| ----------------------------------------- | ----- | ------ | ----- | ----- |
+| STRING                                    |       |        |       |       |
+| ENUM                                      |       |        |       |       |
+| UUID                                      |       |        |       |       |
+| 8 and 16 bit signed INT                   |       |        |       |       |
+| 8, 16, 32, 64 bit unsigned INT            |       |        |       |       |
+| DECIMAL (INT32)                           |       |        |       |       |
+| DECIMAL (INT64)                           |       |        |       |       |
+| DECIMAL (BYTE_ARRAY)                      |       |        |       |       |
+| DECIMAL (FIXED_LEN_BYTE_ARRAY)            |       |        |       |       |
+| DATE                                      |       |        |       |       |
+| TIME (INT32)                              |       |        |       |       |
+| TIME (INT64)                              |       |        |       |       |
+| TIMESTAMP (INT32)                         |       |        |       |       |
+| TIMESTAMP (INT64)                         |       |        |       |       |
+| INTERVAL                                  |       |        |       |       |
```

Comment:
This one doesn't actually exist? (though a number seems to be conventionally reserved for it)

Reply:
It is actually described in the doc: https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#interval. Or is that the description for the ConvertedType?

Reply:
Phew. You're right: https://github.com/apache/parquet-format/blob/e91ab5e892391bd179d3c16b7c1cf4fbaeebbfe7/src/main/thrift/parquet.thrift#L399-L400

Reply:
I did some JIRA diving and found that `INTERVAL` was added by [PARQUET-12](https://issues.apache.org/jira/browse/PARQUET-12) back when converted type and logical type were interchangeable names for the same thing. The separate `LogicalType` struct was added in [PARQUET-906](https://github.com/apache/parquet-format/pull/51) three years later. At that time `INTERVAL` was left out of the new union for [reasons](https://github.com/apache/parquet-format/pull/51#discussion_r118303622)? I'm guessing the description was left in `LogicalTypes.md` to describe the `INTERVAL` converted type. @julienledem can you shed any light here?

Reply:
Some other links:

- The open JIRA ticket about adding (other) interval logical types: https://issues.apache.org/jira/browse/PARQUET-675
- A PR that tried to add this but was eventually closed (didn't read it in detail): https://github.com/apache/parquet-format/pull/165
- And the most recent mailing list message about this: https://lists.apache.org/thread/kpt7q57bz180kzqoc2y294wyb1q20x9j

Reply:
If it is not in the union we might as well call it deprecated and forget it.

Out of curiosity is this type even useful?
- it is 12 bytes which means it will be slower to process than 8 byte types - just like INT96
- given that it stores 3 ints for months, days, millis, it has a range of 357 million years in either direction
- an int64 that stores millis alone has a range of 584 million years

Woot?

Reply:
The trouble is, ConvertedType was originally referred to as "Logical Type" (see https://github.com/apache/parquet-format/blame/b3e928e48a013c7170e93a0cce71a776825cb50d/LogicalTypes.md).  The LogicalType construct in the thrift came along much later.

Reply:
Yes, but my understanding is that this table specifically lists the elements of the `union LogicalType` in thrift, and not of `enum ConvertedType` (and if that is not the case, that should definitely be clarified)

Reply:
My point is, it _is_ a logical type (which is the title of this table), just not in the `LogicalType` union (and as you noted, it's still covered in `LogicalTypes.md`). It's deprecation status is ambiguous. `ConvertedType` is deprecated, and no new types are to be added to it, but readers are still expected to be able to read and understand `ConvertedType`, and writers are still expected to populate it when a mapping exists. All this seven years after deprecation. This is well beyond scope for this PR, but it would be nice if deprecated things actually did disappear from the spec after a certain time (`INT96`, I'm looking at you 👀). Otherwise what's the point of deprecation? Perhaps listing it here, and having few (if any) implementations supporting it would provide the ammunition to kill it off once and for all. 🤷